From 96a1cc731adb28dc4feb71701091b80e67d486a7 Mon Sep 17 00:00:00 2001
From: Wengang Wang <wen.gang.wang@oracle.com>
Date: Tue, 9 Feb 2010 14:57:45 +0800
Subject: [PATCH] ocfs2: Clean up the checks for CoW and direct I/O.
Patch-mainline: 2.6.33
Git-commit: 96a1cc731adb28dc4feb71701091b80e67d486a7

When ocfs2 has to do CoW for refcounted extents, we disable direct I/O
and go through the buffered I/O path.  This makes the combined check
easier to read.

Signed-off-by: Wengang Wang <wen.gang.wang@oracle.com>
Signed-off-by: Joel Becker <joel.becker@oracle.com>
Signed-off-by: Coly Li <coly.li@suse.de>
---
 fs/ocfs2/file.c |    6 ++----
 1 files changed, 2 insertions(+), 4 deletions(-)

Index: linux-2.6.32-sles11-sp1/fs/ocfs2/file.c
===================================================================
--- linux-2.6.32-sles11-sp1.orig/fs/ocfs2/file.c
+++ linux-2.6.32-sles11-sp1/fs/ocfs2/file.c
@@ -1839,6 +1839,8 @@ static int ocfs2_prepare_inode_for_write
 							       &meta_level);
 			if (has_refcount)
 				*has_refcount = 1;
+			if (direct_io)
+				*direct_io = 0;
 		}
 
 		if (ret < 0) {
@@ -1862,10 +1864,6 @@ static int ocfs2_prepare_inode_for_write
 			break;
 		}
 
-		if (has_refcount && *has_refcount == 1) {
-			*direct_io = 0;
-			break;
-		}
 		/*
 		 * Allowing concurrent direct writes means
 		 * i_size changes wouldn't be synchronized, so

